storage/innobase/handler/ha_innodb.cc

```diff
@@ -13084,9 +13098,26 @@ void ha_innobase::adjust_encryption_options(HA_CREATE_INFO *create_info,
   bool is_intrinsic =
       (create_info->options & HA_LEX_CREATE_INTERNAL_TMP_TABLE) != 0;
 
+  if (is_intrinsic) {
+    return;
+  }
+
+  // in case table is first create - create_info will not be updated from
+  // share. Thus we need to update create_info->explicit_encryption here
+  if (!create_info->explicit_encryption &&
+      create_info->used_fields & HA_CREATE_USED_ENCRYPT) {
+#ifdef UNIV_DEBUG
+    ut_ad(table_def->options().exists("explicit_encryption"));
+    bool explicit_encryption{false};
+    table_def->options().get("explicit_encryption", &explicit_encryption);
+    ut_ad(explicit_encryption);
+#endif /* UNIV_DEBUG */
+    create_info->explicit_encryption = true;
+  }
```

Comment:
Looks like DD already has this? can't we do this at server layer?

Reply:
Here we use DD in debug mode to check whether explicit_encryption was set when ENCRYPTION clause was used). On server level were CREATE_INFO is amended like in function: mysql_create_table_no_lock I do not have access to DD. I can move the create_info->explicit_encryption correction to this function and leave the assert here. Or I can leave both here. Which way to go ?

Reply:
Not sure, if you are saying it is not possible to do in server.

How are other create_info attributes filled from DD? We can think of this as attribute not related to storage engines.

If we do at server, other SEs can utilize too.

Extra asserts at SE layer is fine

Reply:
done